src/main/java/spoon/reflect/declaration/CtClass.java

```diff
@@ -44,6 +45,7 @@
 	 * Derived from {@link #getTypeMembers()}
 	 */
 	@DerivedProperty
```

Comment:
remove DerivedProperty?

Reply:
CtClass#getConstructor has already derived property too. So I kept it. But I do not know whether it belongs there or not. I will remove it if you are sure that it has to be removed - no problem.
WDYT @surli @tdurieux

Reply:
getAnonymousExecutables is not a Derived Property. We should remove it.

Reply:
And what about `CtClass#getConstructor` ?

Reply:
`getConstructor(CtTypeReference<?>... parameterTypes)` is not a real getter for me.
It is more a "finder".

Reply:
I am sorry for not exact question. I speak about this one:
```java
       @DerivedProperty
	@PropertyGetter(role = CONSTRUCTOR)
	Set<CtConstructor<T>> getConstructors();
```
it behaves exactly same like `getAnonymousExecutables`, or?

Reply:
This one is not a DerivedProperty for me

Reply:
@monperrus WDYT? You are author of these annotations

Reply:
According to the specification @interface DerivedProperty, `getConstructors()` is a derived property because the content is already scanned by `getTypeMembers()`

Reply:
For me in this case, getTypeMembers() is DerivedProperty and getConstructors, getExecutables, ... are not.
it is just an implementation decision to use getTypeMembers instead of the other methods.
Constructors, methods, ... are the real properties of the medel

Reply:
> Constructors, methods, ... are the real properties of the model

I agree, but the order is defined on the level of typeMember ... so typeMember is real property too.

We need something in between ...

Reply:
We should first try to make a conclusion of the purpose of the CtRole and definition of contract behind the CtRole. I guess each of us understands it different ;-p

Reply:
So after discussion below it is probably clear that @DerivedProperty should stay there. So this PR is correct at this place.